### PR TITLE
Load license from composer metadata

### PR DESCRIPTION
**Pull request by xlgmokha**
Move license detection to a LicensePlugin that applies the license information to each dependency. This change also introduces the `--require` option so that custom plugins can be loaded into the runtime for additional behavior.

### CODE REVIEW

**Comment by xlgmokha** [comment]
@eldemcan Do you mind taking a look?

File: lib/spandx/core/dependency.rb [hunk 1]
```diff
@@ -3,80 +3,46 @@
...
+      attr_reader :package_manager, :name, :version, :licenses, :meta
 
-      attr_reader :package_manager, :name, :version, :meta
-
-      def initialize(package_manager:, name:, version:, meta: {})
+      def initialize(package_manager:, name:, version:, licenses: [], meta: {})
         @package_manager = package_manager
         @name = name
         @version = version
+        @licenses = licenses
```
Comment (eldemcan) [inline lib/spandx/core/dependency.rb:12]: I would consider moving this to `meta`

File: lib/spandx/cli/commands/scan.rb [hunk 2]
```diff
@@ -42,12 +44,21 @@ def each_dependency_from(file)
...
         end
 
         def format(output)
           Array(output).map(&:to_s)
         end
+
+        def enhance(dependency)
+          ::Spandx::Core::Plugin
+            .all
+            .inject(dependency) { |memo, plugin| plugin.enhance(memo) }
```
Comment (eldemcan) [inline lib/spandx/cli/commands/scan.rb:60]: if someone wants to add field to Dependency object is that going to be included in the report ? As far as I understand it won't

File: lib/spandx/core/http.rb [hunk 3]
```diff
@@ -3,17 +3,24 @@
...
   module Core
     class Http
-      attr_reader :driver
+      attr_reader :driver, :retries
 
-      def initialize(driver: Http.default_driver)
+      def initialize(driver: Http.default_driver, retries: 3)
         @driver = driver
+        @retries = retries
+        @circuits = Hash.new { |hash, key| hash[key] = Circuit.new }
```
Comment (eldemcan) [inline lib/spandx/core/http.rb:11]: why do we need `Circuit.new` here ? 

later we do 

```ruby 
  @circuits[host] = false
```

**Review by eldemcan — COMMENTED** [review]
@mokhan Have some comments, you can merge as you wish

File: lib/spandx/cli/commands/scan.rb (hunk 2 above)
Comment (xlgmokha) [reply lib/spandx/cli/commands/scan.rb:60]: To add a new field to the report they would need to provide their own implementation of Dependency that implements `#to_h` and `#to_a`. I hope to have an example soon.

File: lib/spandx/core/http.rb (hunk 3 above)
Comment (xlgmokha) [reply lib/spandx/core/http.rb:11]: This is a quick and dirty implementation of a [circuit breaker](https://www.martinfowler.com/bliki/CircuitBreaker.html). As soon as the scan detects a host that we cannot reach the circuit is opened for that host and this class wont waste time trying to connect to the host over and over for each dependency. I came across this while I scanned my full development directory.

**Pull request merged**